bootstrap/postgres/configs.go

```diff
@@ -460,50 +460,58 @@ func (cr configRepository) DisconnectThing(ctx context.Context, channelID, thing
 	return nil
 }
 
-func (cr configRepository) retrieveAll(owner string, filter bootstrap.Filter) (string, []interface{}) {
-	template := `WHERE owner = $1 %s`
-	params := []interface{}{owner}
-	// One empty string so that strings Join works if only one filter is applied.
-	queries := []string{""}
-	// Since owner is the first param, start from 2.
-	counter := 2
+func (cr configRepository) retrieveAll(domainID string, thingIDs []string, filter bootstrap.Filter) (string, []interface{}) {
+	params := []interface{}{}
+	queries := []string{}
+
+	if domainID != "" {
+		params = append(params, domainID)
+		queries = append(queries, fmt.Sprintf("domain_id = $%d", len(params)))
+	}
+
+	if len(thingIDs) != 0 {
+		queries = append(queries, fmt.Sprintf("magistrala_thing IN ('%s')", strings.Join(thingIDs, "','")))
+	}
+
+	// Adjust the starting point for placeholders based on the current length of params
+	counter := len(params) + 1
 	for k, v := range filter.FullMatch {
-		queries = append(queries, fmt.Sprintf("%s = $%d", k, counter))
 		params = append(params, v)
+		queries = append(queries, fmt.Sprintf("%s = $%d", k, counter))
 		counter++
 	}
 	for k, v := range filter.PartialMatch {
-		queries = append(queries, fmt.Sprintf("LOWER(%s) LIKE '%%' || $%d || '%%'", k, counter))
 		params = append(params, v)
+		queries = append(queries, fmt.Sprintf("LOWER(%s) LIKE '%%' || $%d || '%%'", k, counter))
 		counter++
 	}
 
 	f := strings.Join(queries, " AND ")
 
-	return fmt.Sprintf(template, f), params
+	return "WHERE " + f, params
 }
 
-func (cr configRepository) rollback(content string, tx *sqlx.Tx) {
+func (cr configRepository) rollback(tx *sqlx.Tx) {
 	if err := tx.Rollback(); err != nil {
```

Comment:
Please provide `content` string, This helps to identity at which step error has occured during debug.

```suggestion
func (cr configRepository) rollback(content string, tx *sqlx.Tx) {
	if err := tx.Rollback(); err != nil {
```